code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -591,4 +600,147 @@
+						uses -= round(restore_blood/2)
+						to_chat(user,"<span class='warning'>Your blood rites have restored their blood to safe levels!</span>")
+				var/overall_damage = H.getBruteLoss() + H.getFireLoss() + H.getToxLoss() + H.getOxyLoss()
+				if(overall_damage == 0)
+					to_chat(user,"<span class='cult'>That cultist doesn't require healing!</span>")
+				else
+					var/ratio = uses/overall_damage
+					if(target == user)
+						to_chat(user,"<span class='cult'>Your blood healing is far less efficient when used on yourself!</span>")
+						ratio *= 0.5 // Healing is half as effective if you can't perform a full heal
+						uses -= round(overall_damage) // Healing is twice as expensive even if you can still perform the full heal
+					if(ratio>1)
+						ratio = 1
+						uses -= round(overall_damage)
+						H.visible_message("<span class='warning'>[H] is fully-healed by [user]'s blood magic!</span>")
+					else
+						H.visible_message("<span class='warning'>[H] is healed by [user]'s blood magic!</span>")
+						uses = 0
+					ratio *= -1
+					H.adjustOxyLoss((overall_damage*ratio) * (H.getOxyLoss() / overall_damage), 0)
+					H.adjustToxLoss((overall_damage*ratio) * (H.getToxLoss() / overall_damage), 0)
+					H.adjustFireLoss((overall_damage*ratio) * (H.getFireLoss() / overall_damage), 0)
+					H.adjustBruteLoss((overall_damage*ratio) * (H.getBruteLoss() / overall_damage), 0)
+					H.updatehealth()
+					playsound(get_turf(H), 'sound/magic/staff_healing.ogg', 25)
+					new /obj/effect/temp_visual/cult/sparks(get_turf(H))
+					user.Beam(H,icon_state="sendbeam",time=10)
+			else
+				if(H.stat == DEAD)
+					to_chat(user,"<span class='warning'>Their blood has stopped flowing, you'll have to find another way to extract it.</span>")
+					return
+				if(H.cultslurring)
+					to_chat(user,"<span class='danger'>Their blood has been tainted by an even stronger form of blood magic, it's no use to us like this!</span>")
+					return
+				if(H.blood_volume > BLOOD_VOLUME_SAFE)
+					H.blood_volume -= 100
+					uses += 50
+					user.Beam(H,icon_state="drainbeam",time=10)
+					playsound(get_turf(H), 'sound/magic/enter_blood.ogg', 50)
+					H.visible_message("<span class='danger'>[user] has drained some of [H]'s blood!</span>")
+					new /obj/effect/temp_visual/cult/sparks(get_turf(H))
+				else
+					to_chat(user,"<span class='danger'>They're missing too much blood - you cannot drain them further!</span>")
+					return
+		if(isconstruct(target))
+			var/mob/living/simple_animal/M = target
+			var/missing = M.maxHealth - M.health
+			if(missing)
+				if(uses > missing)
+					M.adjustHealth(-missing)
+					M.visible_message("<span class='warning'>[M] is fully-healed by [user]'s blood magic!</span>")
+				else
+					M.adjustHealth(-uses)
+					M.visible_message("<span class='warning'>[M] is healed by [user]'sblood magic!</span>")
+				playsound(get_turf(M), 'sound/magic/staff_healing.ogg', 25)
+				user.Beam(M,icon_state="sendbeam",time=10)
```

Comment:
this is free, and it probably shouldn't be?

Reply:
Will be fixed in the next update.